core/trino-main/src/test/java/io/trino/execution/TestQueryStateMachine.java

```diff
@@ -742,17 +743,17 @@ private static void assertState(QueryStateMachine stateMachine, QueryState expec
         }
     }
 
-    private QueryStateMachine createQueryStateMachine()
+    protected static QueryStateMachine createQueryStateMachine()
     {
         return queryStateMachine().build();
     }
 
-    private QueryStateMachineBuilder queryStateMachine()
+    private static QueryStateMachineBuilder queryStateMachine()
     {
         return new QueryStateMachineBuilder();
     }
 
-    private class QueryStateMachineBuilder
+    private static class QueryStateMachineBuilder
```

Comment:
Probably makes sense to split this class and associated methods into a separate file. What do you think?